app/services/caution_decision_tree.rb

```diff
@@ -1,17 +1,33 @@
 class CautionDecisionTree < BaseDecisionTree
+  # rubocop:disable Metrics/CyclomaticComplexity
   def destination
     return next_step if next_step
 
     case step_name
+    when :caution_type
+      caution_type_route
     when :caution_date
       edit(:under_age)
     when :under_age
       edit(:caution_type)
-    when :caution_type
+    when :conditional_end_date
       # TODO: change when we have next step
-      { controller: '/home', action: :index }
+      home
     else
       raise InvalidStep, "Invalid step '#{as || step_params}'"
     end
   end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  private
+
+  def caution_type_route
```

Comment:
This method can be renamed to `after_caution_type` (or similar) to better express it is when you answer the "caution type" question, that you enter here. Otherwise looks like it is the actual route of the "caution type" question, which is not.

And for now is ok, but next week when I'm in the office we can rework a bit the `disclosure_check&.caution_type&.include? 'conditional'` part, probably exposing a `#conditional?` method in the CautionType value object that respond true/false and not having this logic here 😉